File: back_end_asp.net/Models/Translation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace back_end_ASP.Models
{
    public class Translation
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public long Id { set; get; }
        [Column("message_id")]
        public long MessageId { set; get; } // Required foreign key property
        [Column("translated_content")]
        public string TranslatedContent { set; get; } = null!;

        public string Language { set; get; } = null!;
        public Message Message { set; get; } = null!;
        public Translation() { }

        public Translation(long id, string translatedContent, string language)
        {
            Id = id;
            TranslatedContent = translatedContent;
            Language = language;
        }






    }
}

File: front_end_flutter/lib/models/translation.dart
class Translation {
  final int id;
  final int messageId;
  final String language;
  final String translatedContent;

  Translation(this.id, this.messageId, this.language, this.translatedContent);

  factory Translation.fromJson(Map<String, dynamic> json) {
    var id = json['id'] as int;
    var messageId = json['messageId'] as int;
    var language = json['language'] as String;
    var translatedContent = json['translatedContent'] as String;
    return Translation(id, messageId, language, translatedContent);
  }

}
